Bukkit/src/main/java/com/plotsquared/bukkit/listener/SpigotListener.java

```diff
@@ -0,0 +1,37 @@
+package com.plotsquared.bukkit.listener;
+
+import com.plotsquared.bukkit.util.BukkitUtil;
+import com.plotsquared.core.location.Location;
+import com.plotsquared.core.plot.Plot;
+import com.plotsquared.core.plot.flag.FlagContainer;
+import com.plotsquared.core.plot.flag.implementations.BeaconEffectsFlag;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityPotionEffectEvent;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Fallback listener for paper events on spigot
+ */
+public class SpigotListener implements Listener {
+
+    @EventHandler(ignoreCancelled = true)
+    public void onEffect(@NotNull EntityPotionEffectEvent event) {
+        if (event.getCause() != EntityPotionEffectEvent.Cause.BEACON) return;
```

Comment:
For P2 codestyle this should have braces

Reply:
Oh sorry. Always do this in my own projects and tried to change it for PlotSquared. Have fallen into old habits there. Will change it